pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2195,6 +2195,15 @@ public void checkGC() {
             CompletableFuture<Void> replCloseFuture = new CompletableFuture<>();
 
             if (TopicName.get(topic).isGlobal()) {
+                // topics with remote (replication) producer should be skipped
+                if (hasRemoteProducers()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("[{}] Global topic has connected remote producers. Not a candidate for GC",
+                                topic);
+                    }
+                    return;
+                }
+
                 // For global namespace, close repl producers first.
```

Comment:
From this comment, it seems we have to close the repl producer first. Otherwise, in a two-way replication scenario. We can get to the "deadlock" and the topic is never deleted.
I'm not sure if it's your expected behaviour or if I'm missing some replicator close logic.